testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -346,19 +346,26 @@ public void testTrinoCreatesSparkDrops()
     @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS}, dataProvider = "storageFormats")
     public void testSparkReadsTrinoPartitionedTable(StorageFormat storageFormat)
     {
-        String baseTableName = "test_spark_reads_trino_partitioned_table_" + storageFormat;
+        String baseTableName = "test_spark_reads_trino_partitioned_table_5" + storageFormat;
         String trinoTableName = trinoTableName(baseTableName);
         String sparkTableName = sparkTableName(baseTableName);
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + trinoTableName);
 
-        onTrino().executeQuery(format("CREATE TABLE %s (_string VARCHAR, _bigint BIGINT) WITH (partitioning = ARRAY['_string'], format = '%s')", trinoTableName, storageFormat));
-        onTrino().executeQuery(format("INSERT INTO %s VALUES ('a', 1001), ('b', 1002), ('c', 1003)", trinoTableName));
+        onTrino().executeQuery(format("CREATE TABLE %s (_string VARCHAR, _varbinary VARBINARY, _bigint BIGINT) WITH (partitioning = ARRAY['_string', '_varbinary'], format = '%s')", trinoTableName, storageFormat));
+        onTrino().executeQuery(format("INSERT INTO %s VALUES ('a', X'0ff102f0feff', 1001), ('b', X'0ff102f0fefe', 1002), ('c', X'0ff102fdfeff', 1003)", trinoTableName));
+
+        Row row1 = row("b", new byte[]{15, -15, 2, -16, -2, -2}, 1002);
+        String selectByString = "SELECT * FROM %s WHERE _string = 'b'";
+        assertThat(onTrino().executeQuery(format(selectByString, trinoTableName)))
+                .containsOnly(row1);
+        assertThat(onSpark().executeQuery(format(selectByString, sparkTableName)))
+                .containsOnly(row1);
+
+        Row row2 = row("a", new byte[]{15, -15, 2, -16, -2, -1}, 1001);
+        String selectByVarbinaryTrino = "SELECT * FROM %s WHERE _varbinary = X'0ff102f0feff'"; //for now this fails on spark see https://githubmemory.com/repo/apache/iceberg/issues/2934
```

Comment:
link to https://github.com/apache/iceberg/issues/2934 instead

selectByVarbinaryTrino ->  selectByVarbinary
(otherwise commenting that "trino select doesn't work in spark" is weird)

also, add `assertQueryFailure`

Reply:
> also, add `assertQueryFailure`

... with onSpark()

bump